src/shared.c

```diff
@@ -200,32 +200,15 @@ void PerpendicularVector(const vec3_t in, vec3_t out) {
  * @brief Projects the normalized directional vectors on to the normal's plane.
  * The fourth component of the resulting tangent vector represents sidedness.
  */
-void TangentVectors(const vec3_t normal, const vec3_t sdir, const vec3_t tdir, vec4_t tangent,
+void TangentVectors(const vec3_t normal, const vec3_t sdir, const vec3_t tdir, vec3_t tangent,
                     vec3_t bitangent) {
 
-	vec3_t s, t;
-
 	// normalize the directional vectors
-	VectorCopy(sdir, s);
-	VectorNormalize(s);
-
-	VectorCopy(tdir, t);
-	VectorNormalize(t);
-
-	// project the directional vector onto the plane
-	VectorMA(s, -DotProduct(s, normal), normal, tangent);
+	VectorCopy(sdir, tangent);
 	VectorNormalize(tangent);
 
-	// resolve sidedness, encode as fourth tangent component
-	CrossProduct(normal, tangent, bitangent);
-
-	if (DotProduct(t, bitangent) < 0.0) {
-		tangent[3] = -1.0;
-	} else {
-		tangent[3] = 1.0;
-	}
-
-	VectorScale(bitangent, tangent[3], bitangent);
+	VectorCopy(tdir, bitangent);
+	VectorNormalize(bitangent);
```

Comment:
I think we have a VectorNormalize2 that could replace the VectorNormalize && VectorCopy. I also wonder if it's time to just kill this function, if that's all it does.